### ISSUE TITLE
Static member and function

### ISSUE THREAD

**Issue by Udayv56** [Udayv56/CPP#12]
**CODE-**
#include <iostream>

using namespace std;

//static data member
class Emp{
    int id;
    static int count; //by default value is 0
    public:
        void setData(void){
            cout<<"Enter the id ";
            cin>>id;
            count++;
        }
        void getData(void){
            cout<<"The id of employee is "<<id<<" and this is employee no. "<<count<<endl;
        }
        // static function can access only static data member
        static void getCount(void){
            //cout<<id ; not allowed
            cout<<"The value of count is "<<count<<endl;
        }
};

int Emp :: count; //here value can be omitted count = 100000 if wanted
int main()
{
    Emp uday,hemant,sangeeta;
    uday.setData();
    uday.getData();
    Emp::getCount();
    hemant.setData();
    hemant.getData();
    Emp::getCount();
    sangeeta.setData();
    sangeeta.getData();
    Emp::getCount();
    return 0;
}

**OUTPUT-**

Enter the id 1
The id of employee is 1 and this is employee no. 1
The value of count is 1
Enter the id 2
The id of employee is 2 and this is employee no. 2
The value of count is 2
Enter the id 3
The id of employee is 3 and this is employee no. 3
The value of count is 3

In the code above, the `Emp` class has a static data member `count`, which is initialized to zero by default. The `count` variable is shared by all objects of the `Emp` class and can be accessed using the scope resolution operator (::) even when no object of the class is created. 

In the `setData` member function, the `count` variable is incremented each time a new object is created. The `getData` member function displays the `id` of the employee along with the current value of the `count` variable.

The `getCount` function is a static member function of the `Emp` class. Static member functions can access only static data members of the class. In this case, the `getCount` function displays the current value of the `count` variable.

In terms of memory allocation, static data members are stored separately in memory, and there is only one copy of the variable that is shared by all objects of the class. The `count` variable is allocated memory once and shared by all objects of the `Emp` class.

Static member functions are also stored separately in memory and can be accessed using the scope resolution operator without creating an object of the class. In this code, the `getCount` function is a static member function that is used to display the current value of the `count` variable without creating an object of the `Emp` class.

In summary, static data members are shared by all objects of the class, and there is only one copy of the variable stored in memory. Static member functions can access only static data members of the class and can be accessed using the scope resolution operator without creating an object of the class.
